cmd/otel-allocator/watcher/promOperator.go

```diff
@@ -297,7 +297,12 @@ func (w *PrometheusCRWatcher) addStoreAssetsForServiceMonitor(
 	for i, endp := range endps {
 		objKey := fmt.Sprintf("serviceMonitor/%s/%s/%d", smNamespace, smName, i)
 
-		if err = store.AddBearerToken(ctx, smNamespace, endp.BearerTokenSecret, objKey); err != nil {
+		// nolint: staticcheck
+		cred := endp.BearerTokenSecret
+		if endp.Authorization != nil && endp.Authorization.Credentials != nil {
+			cred = endp.Authorization.Credentials
```

Comment:
@jaronoff97 I am not sure if this is correct handling of this

Reply:
@pavolloffay all of this is going to go away with the change [here](https://github.com/open-telemetry/opentelemetry-operator/pull/2328/files) so i think we may just want to hold off until that PR is merged.